### PR TITLE
chore: refactor `wrangler login/logout/whoami` to use defineCommand

### PR DESCRIPTION
**Pull request by emily-shen**
Fixes #N/A
Landing #6655 in smaller chunks. This one does `wrangler login/logout/whoami` . There should be no functionality change.

---

- Tests
  - [ ] TODO (before merge)
  - [ ] Tests included
  - [x] Tests not necessary because: no behaviour change, just needs to pass existing
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [ ] Required
  - [x] Not required because:  no behaviour change, just needs to pass existing
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [x] Documentation not necessary because:  internal refactor

### CODE REVIEW

File: packages/wrangler/src/user/commands.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+});
+
+defineCommand({
+	command: "wrangler logout",
+	metadata: {
+		description: "🚪 Logout from Cloudflare",
+		owner: "Workers: Authoring and Testing",
+		status: "stable",
+	},
+	args: {},
```
Comment (penalosa) [inline packages/wrangler/src/user/commands.ts:70]: Can this be omitted entirely with no args?

File: packages/wrangler/src/user/commands.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	async handler(args) {
+		await logout();
+		const config = readConfig(undefined, args);
```
Comment (penalosa) [inline packages/wrangler/src/user/commands.ts:73]: I know this is existing code, but this should probably be `readConfig(args.config, args)` as it is in `login`

File: packages/wrangler/src/user/commands.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+	args: {
+		account: {
+			type: "string",
+			describe:
+				"Show membership information for the given account (id or name).",
+		},
+	},
+	async handler(args) {
+		await whoami(args.account);
+		const config = readConfig(undefined, args);
```
Comment (penalosa) [inline packages/wrangler/src/user/commands.ts:96]: As above

**Review by penalosa — APPROVED** [review]
Couple small things, but otherwise ✅

File: packages/wrangler/src/core/define-command.ts [hunk 4]
```diff
@@ -101,13 +101,20 @@ export type CommandDefinition<
 		 * @default true
 		 */
 		printBanner?: boolean;
+
+		/**
+		 * By default, wrangler will print warnings about wrangler.toml configuration.
+		 * Set this value to `false` to skip printing these warnings.
+		 */
+		printConfigWarnings?: boolean;
```
Comment (edmundhung) [inline packages/wrangler/src/core/define-command.ts:109]: I was rebasing the changes on #6756 over to this PR and notice the same changes doesn't work here. The reason was that `defineCommand` will read the config for you and there was no way to disable the warning.  You can find the full changes that enable this option in https://github.com/cloudflare/workers-sdk/pull/7150/commits/ec6d4cd46074138990400971432afa90566229f6

cc: @penalosa @emily-shen

**Pull request merged**